tests/src/test/scala/mainecoon/tests/autoFunctorTests.scala

```diff
@@ -18,55 +18,39 @@ package mainecoon
 package tests
 
 
-import cats.laws.discipline.SerializableTests
-import cats.laws.discipline.FunctorTests
-import autoFunctorTests._
 import cats.Functor
+import cats.instances.function._
 import cats.kernel.Eq
-import org.scalacheck.{Arbitrary, Cogen}
+import cats.kernel.instances.string._
+import cats.kernel.instances.tuple._
 import cats.laws.discipline.eq._
+import cats.laws.discipline.{FunctorTests, SerializableTests}
+import cats.syntax.arrow._
+import mainecoon.tests.autoFunctorTests._
+import org.scalacheck.{Arbitrary, Cogen}
 
 class autoFunctorTests extends MainecoonTestSuite {
 
-  checkAll("SimpleAlg[Option]", FunctorTests[SimpleAlg].functor[Float, String, Int])
-  checkAll("Functor[SimpleAlg]", SerializableTests.serializable(Functor[SimpleAlg]))
-
-  test("non effect method correctly delegated") {
-    val doubleAlg = AlgWithNonEffectMethodFloat.map(_.toDouble)
-    doubleAlg.bar("big") should be("big")
-    doubleAlg.foo("big") should be(3d)
-  }
+  checkAll("SimpleAlg[Option]", FunctorTests[TestAlgebra].functor[Long, String, Int])
+  checkAll("Functor[SimpleAlg]", SerializableTests.serializable(Functor[TestAlgebra]))
 
   test("extra type param correctly handled") {
     val doubleAlg = AlgWithExtraTypeParamFloat.map(_.toDouble)
     doubleAlg.foo("big") should be(3d)
   }
-
-  test("Alg with non effect method with default Impl") {
-    val tryInt = new AlgWithDefaultImpl[Int] {
-      def plusOne(i: Int): Int = i + 1
-    }
-
-    tryInt.map(_.toString).plusOne(3) should be("4")
-    tryInt.map(_.toString).minusOne(2) should be(1)
-  }
 }
 
 object autoFunctorTests {
-  @autoFunctor
-  trait SimpleAlg[T] {
-    def foo(a: String): T
-  }
 
   @autoFunctor
-  trait AlgWithNonEffectMethod[T] {
-    def foo(a: String): T
-    def bar(a: String): String
-  }
+  trait TestAlgebra[T] {
+    def abstractEffect(a: String): T
 
-  @autoFunctor
-  trait AlgWithDef[T] {
-    def foo: T
```

Comment:
just noticed that this is not tested before, but at least it compiled. Can we add a def method without arguments in `TestAlgebra` to test this scenario as well?

Reply:
Done